### PR TITLE
Fix broken RST docs

### PR DESCRIPTION
**Pull request by buger**
# Description
Fix broken RST docs, and improve the examples

Address part of https://github.com/open-telemetry/opentelemetry-python-contrib/issues/3061

## Type of change

Please delete options that are not relevant.

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

# How Has This Been Tested?
Build docs locally and validated rendering

# Does This PR Require a Core Repo Change?

- [ ] Yes. - Link to PR: 
- [x] No.

# Checklist:

See [contributing.md](https://github.com/open-telemetry/opentelemetry-python-contrib/blob/main/CONTRIBUTING.md) for styleguide, changelog guidelines, and more.

- [ ] Followed the style guidelines of this project
- [ ] Changelogs have been updated
- [ ] Unit tests have been added
- [x] Documentation has been updated

### CODE REVIEW

**Review by xrmx — APPROVED** [review]

**Review by emdneto — CHANGES_REQUESTED** [review]
Thanks for the PR. I think even with that fix, we'll still face broken RSTs because the actual docs come from [here](https://github.com/open-telemetry/opentelemetry-python-contrib/blob/main/instrumentation/opentelemetry-instrumentation-confluent-kafka/src/opentelemetry/instrumentation/confluent_kafka/__init__.py#L21) for example. So, the correct fix should be in the respective doc at __init__.py of each package.

**Comment by buger** [comment]
@emdneto done!

File: instrumentation/opentelemetry-instrumentation-aiokafka/src/opentelemetry/instrumentation/aiokafka/__init__.py [hunk 1]
```diff
@@ -45,7 +45,8 @@ def async_produce_hook(span: Span, args, kwargs)
 def async_consume_hook(span: Span, record: kafka.record.ABCRecord, args, kwargs)
 for example:
```
Comment (emdneto) [inline instrumentation/opentelemetry-instrumentation-aiokafka/src/opentelemetry/instrumentation/aiokafka/__init__.py:46]: I think we can remove from L38 to L46 here.

File: instrumentation/opentelemetry-instrumentation-confluent-kafka/src/opentelemetry/instrumentation/confluent_kafka/__init__.py [hunk 2]
```diff
@@ -54,7 +54,6 @@ def basic_consume_loop(consumer, topics):
             consumer.close()
 
     basic_consume_loop(consumer, "my-topic")
-    ---
 
 The _instrument method accepts the following keyword args:
```
Comment (emdneto) [inline instrumentation/opentelemetry-instrumentation-confluent-kafka/src/opentelemetry/instrumentation/confluent_kafka/__init__.py:58]: Same here. From L59 to L71

File: instrumentation/opentelemetry-instrumentation-kafka-python/src/opentelemetry/instrumentation/kafka/__init__.py [hunk 3]
```diff
@@ -45,7 +45,8 @@ def produce_hook(span: Span, args, kwargs)
 def consume_hook(span: Span, record: kafka.record.ABCRecord, args, kwargs)
 for example:
```
Comment (emdneto) [inline instrumentation/opentelemetry-instrumentation-kafka-python/src/opentelemetry/instrumentation/kafka/__init__.py:46]: Same here from L38 to L46

**Review by xrmx — APPROVED** [review]
I think we can get the syntax fixes now and revise the content later

**Review by emdneto — APPROVED** [review]

**Pull request merged**